### PR TITLE
Fix #2211: Send API key in DAU ping

### LINKED ISSUE #2211: Internal Issue #678
### Description: 
Pls implement https://github.com/brave/internal/issues/678

### PR DESCRIPTION
**Pull request by kylehickinson**
Refs brave/internal#678

## Summary of Changes

Fixes #2211 

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Reviewer Checklist:

- [x] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `release-notes/(include|exclude)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [x] Adequate unit test coverage exists to prevent regressions.
- [x] Adequate test plan exists for QA to validate (if applicable).
- [x] Issue is assigned to a milestone (should happen at merge time).

### CODE REVIEW

**Review by iccub — CHANGES_REQUESTED** [review]
Write unit test for it, can be super simple just to verify the param is attached correctly and that key matches the one in debug plist

**Comment by kylehickinson** [comment]
Potential changes to where the key is placed, so blocking for now until decision is made

**Review by iccub — APPROVED** [review]
approved, would be good if @jhreis could take a final look at it

also the server side must respect the new header only for iOS1.15 and newer builds, is this going to be implemented?

File: BraveShared/Analytics/DAU.swift [hunk 1]
```diff
@@ -142,7 +152,13 @@ public class DAU {
         
         let lastPingTimestamp = [Int((today).timeIntervalSince1970)]
         
-        return ParamsAndPrefs(queryParams: params, lastLaunchInfoPreference: lastPingTimestamp)
+        var headers: [String: String] = [:]
+        
+        if let key = self.apiKey, !key.isEmpty {
+            headers["x-brave-api-key"] = key
```
Comment (diracdeltas) [inline BraveShared/Analytics/DAU.swift:158]: this should be added to https://github.com/brave/brave-browser/wiki/Custom-Headers

Comment (kylehickinson) [reply BraveShared/Analytics/DAU.swift:158]: @diracdeltas As far as I know this isn't for any websites, just for the DAU ping, so not sure if we want to add it to that custom headers wiki

**Review by jhreis — APPROVED** [review]

**Pull request merged**